build.sbt

```diff
@@ -995,15 +990,27 @@ def genSmithyImpl(config: Configuration) = Def.task {
           ) { case ((changed, files), outputs) =>
             if (changed || outputs.isEmpty) {
               val inputs = inputFiles.map(_.getAbsolutePath()).toList
-              val args =
-                List("--output", outputDir) ++
-                  List("--resource-output", resourceOutputDir) ++
-                  (if (discoverModels) List("--discover-models") else Nil) ++
-                  (if (allowedNS.isDefined)
-                     List("--allowed-ns", allowedNS.get.mkString(","))
-                   else Nil) ++
-                  inputs ++
-                  skip.flatMap(s => List("--skip", s))
+              val outputOpt = List("--output", outputDir)
+              val resourceOutputOpt =
+                List("--resource-output", resourceOutputDir)
+              val discoverModelsOpt =
+                (if (discoverModels) List("--discover-models") else Nil)
+              val allowedNsOpt =
+                if (allowedNS.isDefined)
+                  List("--allowed-ns", allowedNS.get.mkString(","))
+                else Nil
+              val skipOpt = skip.flatMap(s => List("--skip", s))
+              val dependenciesOpt =
+                if (smithy4sDependencies.nonEmpty)
+                  List("--dependencies", smithy4sDependencies.mkString(","))
```

Comment:
added this option

Reply:
I love that you got rid of most genDiscoverModels